### DISCUSSION TITLE
Error Cannot declare class

### DISCUSSION THREAD

**Discussion by nucleus-genius** [PHPMailer/PHPMailer#2658] [Q&A]
Cannot declare class PHPMailer\\PHPMailer\\PHPMailer, because the name is already in use in ....../PHPMailer/PHPMailer.php on line 32

code:

use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\SMTP;

require_once (quasar_form_path.'lib/PHPMailer/PHPMailer.php'); 
require_once  (quasar_form_path.'lib/PHPMailer/SMTP.php');

$mail = new PHPMailer();
            try {
                $msg = "ok";
                $mail->isSMTP();   
                $mail->CharSet = "UTF-8";                                          
                $mail->SMTPAuth   = true;
    
                // setting
                $mail->Host       = $form_array_mail[10]; // SMTP 
                $mail->Username   = $form_array_mail[8]; // login
                $mail->Password   = $form_array_mail[9]; // password
                $mail->SMTPSecure = 'ssl'; // ssl
                $mail->Port       = $form_array_mail[11];//port
                $mail->setFrom($form_array_mail[7], $form_array_mail[6]); // email and name 
                //address of the recipient
                //many mails
                if (preg_match("/,/", $form_array_mail[0])) {
                    $massivmail = explode(',' , $form_array_mail[0]);
                    foreach($massivmail as $key => $value){
                        $mail->addAddress($value); 
                   }
                }
                //one mail
                else { 
                    $mail->addAddress($form_array_mail[0]);
                }
                $str = '';
               
                // message
                $mail->isHTML(true);
                $mail->Subject = $form_array_mail[1];//heading message
                $mail->Body    = $messag_email; //message
                
                 
                // sending status
                if ($mail->send()) { echo esc_html( $msg ); wp_mail('info@quasar-media.ru', 'test', 'succes'); }
                else { echo esc_html__('Message was not sent. Your email settings are incorrect','quasar-form');  wp_mail('info@quasar-media.ru', 'test', 'error');}
           } 
           catch (Exception $e) { echo esc_html__('Message was not sent. Cause of error','quasar-form').": {$mail->ErrorInfo}"; }
        }

**Comment by Synchro** [comment]
This will be because you've already loaded it from some other location, for example via composer's autoload.

**Comment by nucleus-genius** [comment]
I didn't upload anything else. The code I showed is all the code related to phpmailer

**Reply by Synchro** [reply]
It really isn't. The error message says it's on line 32, but in your script the class is loaded on line 4. This suggests there is other code. You're using a constant in the require line without having defined it. Then you removed the full path to the conflicting class shown in the error message. If you remove critical information from your post, it's going to prevent anyone from diagnosing it effectively.

**Reply by nucleus-genius** [reply]
full text of the error:
[Sun Mar 27 00:26:51 2022] [error] [client 80.95.44.200:0] PHP Fatal error:  Cannot declare class PHPMailer\\PHPMailer\\PHPMailer, because the name is already in use in /home/r/rocketq7/free-form/public_html/wp-includes/PHPMailer/PHPMailer.php on line 32, referer http://rocketq7.beget.tech/

const:
define( 'quasar_form_path', plugin_dir_path(__FILE__ ) );

**Reply by Synchro** [reply]
OK, so you have PHPMailer being loaded from two different places. It looks like the other one is WordPress' own copy, so all you should have to do is delete or comment out the require lines in your script. Since you're apparently in WordPress, you might want to look into making direct use of WordPress' built-in mail functionality (the `wp_mail` function), which you can customise quite easily, [as described here](https://github.com/PHPMailer/PHPMailer/issues/2221#issuecomment-744040574).

**Reply by nucleus-genius** [reply]
I have never heard that the phpmailer library is part of wordpress. In my opinion, this is not so. I am using phpmailer to send emails via SMTP.

**Reply by Synchro** [reply]
That might be your opinion, but [it's wrong](https://build.trac.wordpress.org/browser/trunk/wp-includes/PHPMailer).

**Comment by nucleus-genius** [comment]
The problem disappeared after I replaced `$mail = new PHPMailer()` to `$mail = new PHPMailer\PHPMailer\PHPMailer();`
